### PR TITLE
Make sqlite default db

### PR DESCRIPTION
**Pull request by sdc50**
Changes the default database configuration in settings to use SQLite instead of Postgres. 

Provides backwards compatibility code to detect if a default Postgres database was already configured to preserves the `postgresql` engine in that case.

### CODE REVIEW

**Comment by sdc50** [comment]
Remaining work for this PR:

- [ ] Add data migration command/instructions to switch from one type of DB to another
- [ ] Update the documentation for database settings
- [ ] Update deployment documentation

see: https://github.com/tethysplatform/tethys/issues/925#issuecomment-1450384795

**Comment by sdc50** [comment]
@swainn  I think this is ready to go. As for the production documentation I think that it is out of scope to write a detailed guide for setting up other types of databases, and if we ever did I would want to throughly test them in a production environment first. What I did include is an explanation that other database backends are possible and referred to the Django Database docs for more details.

**Comment by sdc50** [comment]
Also, it looks like the most recent version of `python-social-auth` was breaking our tests. As far as I can tell it just dropped support for a very outdated way of using middleware, but our tests were still calling it with the old style (not passing in a `get_request` argument).

**Comment by swainn** [comment]
> Also, it looks like the most recent version of `python-social-auth` was breaking our tests. As far as I can tell it just dropped support for a very outdated way of using middleware, but our tests were still calling it with the old style (not passing in a `get_request` argument).

Is it only our tests that are using the old way? Is the code up-to-date?

File: tests/unit_tests/test_tethys_portal/test_middleware.py [hunk 1]
```diff
@@ -22,13 +22,14 @@ def tearDown(self):
...
     ):
+        mock_get_response = mock.MagicMock()
         mock_request = mock.MagicMock()
         mock_exception = mock.MagicMock()
         mock_hasattr.return_value = True
         mock_isinstance.return_value = True
         mock_request.user.is_anonymous = True
 
-        obj = TethysSocialAuthExceptionMiddleware()
+        obj = TethysSocialAuthExceptionMiddleware(mock_get_response)
```
Comment (swainn) [inline tests/unit_tests/test_tethys_portal/test_middleware.py:32]: Is there a corresponding change we need to make in the `TethysSocialAuthExceptionMiddleware`?

File: tethys_cli/db_commands.py [hunk 2]
```diff
@@ -33,18 +33,22 @@ def add_db_parser(subparsers):
         "command",
         help="Performs operations on the Tethys Database.\n"
         "  * init - Creates a new, locally running PostgreSQL database server.\n"
-        "  * start - Starts the local database server.\n"
-        "  * stop - Stops the local database server.\n"
-        "  * status - Gets the current status of the local database server.\n"
-        "  * create - Creates the Tethys Portal database on the database server configured "
+        "  * start - Starts the local (PostgreSQL) database server.\n"
+        "  * stop - Stops the local (PostgreSQL) database server.\n"
+        "  * status - Gets the current status of the local (PostgreSQL) database server.\n"
```
Comment (swainn) [inline tethys_cli/db_commands.py:38]: "the local (Postgresql)" -> "a local Postgresql"

**Review by swainn — COMMENTED** [review]
Do we need to add sqlite3 to the environment.yml? Are we planning to take out the postgresql dependencies and add instructions for installing them when you want to use postgresql?

File: tethys_cli/db_commands.py (hunk 2 above)
Comment (swainn) [reply tethys_cli/db_commands.py:38]: *PostgreSQL

**Comment by sdc50** [comment]
> Do we need to add sqlite3 to the environment.yml? Are we planning to take out the postgresql dependencies and add instructions for installing them when you want to use postgresql?

sqlite3 is native to Python so no dependencies needed. I wasn't planning to remove the PostgreSQL dependencies. I think they are still handy to have available to test production environments.

File: tests/unit_tests/test_tethys_portal/test_middleware.py (hunk 1 above)
Comment (sdc50) [reply tests/unit_tests/test_tethys_portal/test_middleware.py:32]: No, as far as I can tell it was already compatible with the newer way of calling it (passing in the get_response). The latest change just dropped support for the old way of calling it (without the get_response arg). The only change necessary was to update our tests to use the new style.

File: tethys_cli/db_commands.py [hunk 3]
```diff
@@ -33,18 +33,22 @@ def add_db_parser(subparsers):
         "command",
         help="Performs operations on the Tethys Database.\n"
         "  * init - Creates a new, locally running PostgreSQL database server.\n"
-        "  * start - Starts the local database server.\n"
-        "  * stop - Stops the local database server.\n"
-        "  * status - Gets the current status of the local database server.\n"
-        "  * create - Creates the Tethys Portal database on the database server configured "
+        "  * start - Starts the local (PostgreSQL) database server.\n"
```
Comment (sdc50) [inline tethys_cli/db_commands.py:36]:
```suggestion
        "  * start - Starts a local PostgreSQL database server.\n"
```

File: tethys_cli/db_commands.py [hunk 4]
```diff
@@ -33,18 +33,22 @@ def add_db_parser(subparsers):
...
+        "  * stop - Stops the local (PostgreSQL) database server.\n"
```
Comment (sdc50) [inline tethys_cli/db_commands.py:37]:
```suggestion
        "  * stop - Stops a local PostgreSQL database server.\n"
```

File: tethys_cli/db_commands.py (hunk 2 above)
Comment (sdc50) [inline tethys_cli/db_commands.py:38]:
```suggestion
        "  * status - Gets the current status of a local PostgreSQL database server.\n"
```

**Comment by sdc50** [comment]
> Is it only our tests that are using the old way? Is the code up-to-date?

The `TethysSocialAuthExceptionMiddleware` class is never actually instantiated in our code.  It's added as a middleware, so I think it's only instantiated by Django.

**Review by swainn — APPROVED** [review]

**Pull request merged**